db/commitlog/commitlog.cc

```diff
@@ -1799,6 +1805,16 @@ future<> db::commitlog::segment_manager::delete_segments(std::vector<sstring> fi
             clogger.error("Could not delete segment {}: {}", filename, std::current_exception());
         }
     }
+
+    // #8376 - if we had an error in recycling (disk rename?), and no elements
+    // are available, we could have waiters hoping they will get segements.
+    // abort the queue (not permanent, but wakes up waiters), and let them
+    // retry. Since we did deletions instead, disk footprint should allow
+    // for new allocs at least. Or more likely, everything is broken, but
+    // we will at least make more noise.
+    if (recycle_error && _recycled_segments.empty()) {
+        _recycled_segments.abort(recycle_error);
```

Comment:
I'm afraid that [`seastar::queue<T>::abort()`](https://github.com/scylladb/seastar/blob/72e3baed9c3657d49d9f1b82045831e72b3dc2a4/include/seastar/core/queue.hh#L119) doesn't work in the way you described. It not only sends the error to the current waiters, but prevents further read/write operations on that queue. Notice the comment above the definition of abort:

```c++
/// Destroy any items in the queue, and pass the provided exception to any
/// waiting readers or writers - or to any later read or write attempts.
```

Looking at the code, this seems to be enforced in some of methods of the queue that the commitlog uses, e.g. `not_empty` and `consume`.

Reply:
Right, sorry. Forgot the reset of queue. Updated.